fluentforms/core/src/main/java/com/_4point/aem/fluentforms/impl/assembler/SafeDocAssemblerServiceAdapterWrapper.java

```diff
@@ -0,0 +1,49 @@
+package com._4point.aem.fluentforms.impl.assembler;
+
+import java.util.Map;
+import java.util.Objects;
+
+import com._4point.aem.fluentforms.api.Document;
+import com._4point.aem.fluentforms.api.assembler.AssemblerOptionsSpec;
+import com._4point.aem.fluentforms.api.assembler.AssemblerResult;
+import com._4point.aem.fluentforms.api.assembler.AssemblerService.AssemblerServiceException;
+import com.adobe.fd.assembler.client.OperationException;
+import com.adobe.fd.assembler.client.PDFAConversionOptionSpec;
+import com.adobe.fd.assembler.client.PDFAConversionResult;
+import com.adobe.fd.assembler.client.PDFAValidationOptionSpec;
+import com.adobe.fd.assembler.client.PDFAValidationResult;
+
+
+public class SafeDocAssemblerServiceAdapterWrapper implements TraditionalDocAssemblerService {
+	private final TraditionalDocAssemblerService assemblerService;
+
+	public SafeDocAssemblerServiceAdapterWrapper(TraditionalDocAssemblerService docAssemblerService) {
+		super();
+		this.assemblerService = docAssemblerService;
+	}
+
+	@Override
+	public AssemblerResult invoke(Document ddx, Map<String, Object> sourceDocuments,
+			AssemblerOptionsSpec adobAssemblerOptionSpec) throws AssemblerServiceException, OperationException {
+		Objects.requireNonNull(ddx, "ddx Document cannot be null.");
+		Objects.requireNonNull(sourceDocuments, "sourceDocuments cannot be null.");
+		return assemblerService.invoke(ddx, sourceDocuments, adobAssemblerOptionSpec);
+
+	}
+
+	@Override
+	public PDFAValidationResult isPDFA(Document inDoc, PDFAValidationOptionSpec options) {		
+		return null;
+	}
```

Comment:
I'd rather throw an UnsupportedOperationException here rather than returning null.  Returning null is probably going to lead to a NullPointerException somewhere else in the code.  Backtracking to determine where the null came from requires some work.  Throwing an Exception is more direct and is easier to trace back to the source.

Reply:
Returning adobe api object here